src/otp-field.tsx

```diff
@@ -0,0 +1,309 @@
+    );
+
+    React.useEffect(() => {
+        // https://developer.mozilla.org/en-US/docs/Web/API/WebOTP_API
+        if ('OTPCredential' in window && !isWaitingForSms) {
+            isWaitingForSms = true;
+            const abortController = new AbortController();
+            navigator.credentials
+                .get({
+                    // @ts-expect-error: otp is not in the types yet
+                    otp: {transport: ['sms']},
+                    signal: abortController.signal,
+                })
+                .then((otp) => {
+                    if (otp) {
+                        // @ts-expect-error: otp is not in the types yet
+                        const code = otp.code.slice(0, length);
+                        changeValue(code);
+                    }
+                })
+                .catch(() => {
+                    // ignore;
+                });
+            return () => {
+                isWaitingForSms = false;
+                abortController.abort();
+            };
+        }
+    }, [changeValue, length]);
+
+    const createInputChangeHandler = (index: number) => (event: React.ChangeEvent<HTMLInputElement>) => {
+        const eventValue = event.target.value;
+
+        // digit was deleted
+        if (eventValue === '') {
+            // case already handled in onKeyDown
+            return;
+        }
+
+        const currentValue = controlledValue[index];
+        let newInputValue: string = eventValue;
+        if (!currentValue || currentValue === eventValue) {
+            newInputValue = eventValue;
+        } else if (currentValue === eventValue[0]) {
+            newInputValue = eventValue.slice(1);
+        } else if (currentValue === eventValue[eventValue.length - 1]) {
+            newInputValue = eventValue.slice(0, -1);
+        }
+
+        let indexToFocus = index;
+        let newControlledValue = controlledValue;
+
+        // in the case of an autocomplete or copy and paste
+        if (newInputValue.length >= 2) {
+            const toPaste = newInputValue.slice(0, length - index);
+            const prevChars = controlledValue.slice(0, index);
+
+            newControlledValue = prevChars + toPaste;
+            indexToFocus = Math.min(index + toPaste.length, length - 1);
+        } else {
+            newControlledValue =
+                controlledValue.slice(0, index) + newInputValue + controlledValue.slice(index + 1);
+            indexToFocus = index + 1;
+        }
+
+        // need to flush sync to be commit the new values to the dom before changing the focus
+        flushSync(() => {
+            changeValue(newControlledValue);
+        });
+        if (indexToFocus !== index && indexToFocus <= length - 1) {
+            inputsList[indexToFocus]?.focus();
+        }
+    };
+
+    return (
+        <Inline space={8}>
+            {Array.from({length}).map((_, index) => (
+                <div
+                    key={index}
```

Comment:
this gets reported as an error in SonarQube :/

Reply:
jajaja, I won't do that